tests/test_realisation.py

```diff
@@ -0,0 +1,403 @@
+# NOTE: this fix contains tests that mirror the Realisations wiki. If
+# these tests fail, you should update the wiki if necesary to ensure
+# it stays consistent with the codebase.
+import json
+from pathlib import Path
+
+import numpy as np
+import pytest
+import schema
+from velocity_modelling import bounding_box
+
+from source_modelling import rupture_propagation
+from workflow import defaults, realisations
+
+
+def test_bounding_box_example(tmp_path: Path):
+    domain_parameters = realisations.DomainParameters(
+        resolution=0.1,  # a 0.1km resolution
+        domain=bounding_box.BoundingBox.from_centroid_bearing_extents(
+            centroid=np.array([-43.53092, 172.63701]),
+            bearing=45.0,
+            extent_x=100.0,
+            extent_y=100.0,
+        ),
+        depth=40.0,
+        duration=60.0,
+        dt=0.005,
+    )
+    realisation_ffp = tmp_path / "realisation.json"
+    domain_parameters.write_to_realisation(realisation_ffp)
+    with open(realisation_ffp, "r") as realisation_handle:
+        assert json.load(realisation_handle) == {
+            "domain": {
+                "resolution": 0.1,
+                "domain": [
+                    {"latitude": -43.524793866326725, "longitude": 171.76204128885567},
+                    {"latitude": -44.16756820707226, "longitude": 172.63312824122775},
+                    {"latitude": -43.53034935969409, "longitude": 173.51210368762364},
+                    {"latitude": -42.894200350955856, "longitude": 172.64076673694242},
+                ],
+                "depth": 40.0,
+                "duration": 60.0,
+                "dt": 0.005,
+            }
+        }
+    domain_parameters_read = realisations.DomainParameters.read_from_realisation(
+        realisation_ffp
+    )
+    assert domain_parameters_read.depth == domain_parameters.depth
```

Comment:
Could these be done like the ones below and just compare the to_dict forms? Also is the name of this function right? I feel like it should be test_domain_parameters()

Reply:
Yes I could probably refactor the test cases to use the to_dict form for most things, except for inferred properties like `nx` and `ny` which should be tested explicitly because they are not saved in the realisation and wouldn't be tested by `to_dict`.